tools/file_packager.py

```diff
@@ -707,6 +707,13 @@ def generate_js(data_target, data_files, metadata):
             Module['removeRunDependency']('datafile_%s');''' % (meta, "true" if options.use_preload_plugins else "false", js_manipulation.escape_for_js_string(data_target))
 
     package_uuid = uuid.uuid4()
+
+    # Set the id to a hash of the preloaded data, so that caches survive over multiple builds
+    # if the data has not changed.
+    if options.has_preloaded:
+      with open(data_target, 'rb') as f:
```

Comment:
`data = utils.read_binary(data_target)`

Reply:
How about putting this is in an if/else with the call to uuid4()?

Also, it looks like package_uuid and PACKAGE_UUID are only ever needed when `options.use_preload_cache` so maybe we can put this whole thing inside of `if options.use_preload_cache`?

Reply:
Something like this: https://github.com/emscripten-core/emscripten/pull/16843

Reply:
OK- updated the PR.

Should I also just rename it to PACKAGE_HASH too?